x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/gen/script/ScriptWeaver.java

```diff
@@ -63,6 +65,11 @@ default ScriptTemplate scriptWithFoldable(Expression foldable) {
             return new ScriptTemplate(processScript("{sql}.intervalDayTime({},{})"),
                     paramsBuilder().variable(idt.interval().toString()).variable(idt.dataType().name()).build(),
                     dataType());
+        } else if (fold instanceof OffsetTime) {
```

Comment:
Minor nitpick - all these branches should be `if` not `else if` since the return statement does an early exit.